src/main/kotlin/StringAddition.kt

```diff
@@ -0,0 +1,46 @@
+import util.findPattern
+import util.takeIfNotEmpty
+import util.toIntThrow
+
+class StringAddition(private val exp: String) {
+    fun execute(): Int {
+        val (split, target) = exp.findPattern(CUSTOM_PATTERN).extract(default = exp)
+        return add(target, split)
```

Comment:
```suggestion
        return target
            .takeIfNotEmpty()
            ?.split(addPrefix(delimiter).toRegex())
            ?.map { it.toIntThrow() }
            ?.onEach { require(it >= 0) { "0 또는 양수여야 합니다." } }
            ?.reduce { total, num -> total + num }
            ?: 0
```
1. (좀 과격하지만) fluent api 에 익숙한 조직이라면 애매하게 함수를 나누는것보다 한번에 표현하는것도 좋다고 생각합니다.
2. 현재 `add` 함수는 이름만 봐서는 큰 `target` 과 `split` 값을 append 한다고 느낄수도 있을것 같아요.